### PR TITLE
Guard against empty values for repo and share image & tag names

### PR DESCRIPTION
**Pull request by amohammedalfresco**

### CODE REVIEW

File: templates/acs.yaml [hunk 1]
```diff
@@ -206,6 +206,10 @@ Parameters:
 
 Conditions:
   PrivateRegistry: !Not [!Equals [!Ref RegistryCredentials, ""]]
+  NewRepoImage: !Not [!Equals [!Ref RepoImage, ""]]
```
Comment (gavincornwell) [inline templates/acs.yaml]: Why not use the approach we've used elsewhere of applying a constraint to the parameter definition?

Comment (amohammedalfresco) [reply templates/acs.yaml]: Sure.  I will update the cfn parameter definitions.  Thanks @gavincornwell.

Comment (amohammedalfresco) [reply templates/acs.yaml]: Actually, my original plan was to add these supporting CFN parameters with a default value.  But, then instead chose to leave them empty and condition it.  If these param are not empty then override the values of default helm chart.

But, an issue was noted with empty values, the bash script that sets the optional arg was having problems, it was taking the value of empty string with the next optional parameter.

Hence, I just added some conditions for empty vs non-empty values.

Comment (gavincornwell) [reply templates/acs.yaml]: Yes, I see what you mean, in which case ignore my suggestion/comment ;-)

**Comment by amohammedalfresco** [comment]
Waiting for a green build on bamboo because AWS EKS Worker node got problems today and stacks are failing due to it.

**Pull request merged**